# 라이브러리 불러오기

In [1]:
#드라이브마운트
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import warnings
warnings.filterwarnings('ignore')

In [3]:
import matplotlib.pyplot as plt

plt.rc('font', family='NanumBarunGothic')

In [4]:
import pandas as pd
import numpy as np

In [5]:
!sudo apt-get install -y fonts-nanum
!sudo fc-cache -fv
!rm ~/.cache/matplotlib -rf

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  fonts-nanum
0 upgraded, 1 newly installed, 0 to remove and 38 not upgraded.
Need to get 10.3 MB of archives.
After this operation, 34.1 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fonts-nanum all 20200506-1 [10.3 MB]
Fetched 10.3 MB in 0s (20.8 MB/s)
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 1.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin: 
Selecting previously unselected package fonts-nanum.
(Reading database ... 121752 files and direc

# 데이터 불러오기 및 정보 정리

# C매장 데이터 전처리

In [6]:
df = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/product_data_001.csv',encoding='euc-kr')

In [7]:
df.head()

,Unnamed: 0.1,Unnamed: 0,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,0,0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04,NaN
1,1,1,2023-01-06,1476143616,정상회원,간식,빵,"식빵, 식사대용",아침빵(240g/8개)/매장,1.0,4403,매장C,10:09,NaN
2,2,2,2023-01-06,743192512,정상회원,채소,양념채소,생강,생강(300g),1.0,7083,매장C,10:12,NaN
3,3,3,2023-01-06,176912064,정상회원,축산물,유제품,우유/산양유,유기농저지방우유(종이팩/900ml),1.0,7849,매장C,10:23,NaN
4,4,4,2023-01-06,87592064,정상회원,양념/가루,가루,양념가루,카레(매운맛: 100g),3.0,15507,매장C,10:29,NaN


In [8]:
# Unnamed : 0.1 , Unnamed: 0 불필요한 컬럼으로 삭제
df =df.drop(['Unnamed: 0.1', 'Unnamed: 0'],axis=1)
df.head()

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04,NaN
1,2023-01-06,1476143616,정상회원,간식,빵,"식빵, 식사대용",아침빵(240g/8개)/매장,1.0,4403,매장C,10:09,NaN
2,2023-01-06,743192512,정상회원,채소,양념채소,생강,생강(300g),1.0,7083,매장C,10:12,NaN
3,2023-01-06,176912064,정상회원,축산물,유제품,우유/산양유,유기농저지방우유(종이팩/900ml),1.0,7849,매장C,10:23,NaN
4,2023-01-06,87592064,정상회원,양념/가루,가루,양념가루,카레(매운맛: 100g),3.0,15507,매장C,10:29,NaN


In [9]:
#인덱스 재설정
df=df.reset_index(drop=True)

In [12]:
df['공급일자'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 123191 entries, 0 to 123190
Series name: 공급일자
Non-Null Count   Dtype 
--------------   ----- 
123191 non-null  object
dtypes: object(1)
memory usage: 962.6+ KB


In [13]:
#공급일자에서 구매월 / 구매일 파생변수 생성
df['공급일자']= pd.to_datetime(df['공급일자'])

In [18]:
df['구매월']=df['공급일자'].dt.month
df['구매일']=df['공급일자'].dt.day

In [19]:
df.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,월,일,구매월,구매일
0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04,NaN,1,6,1,6


In [21]:
df=df.drop(['월','일'],axis=1)

In [22]:
df.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일
0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04,NaN,1,6


In [23]:
df['구매월'].unique()

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11])

In [24]:
df['구매일'].unique()

array([ 6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
       23, 24, 25, 26, 27, 28, 29, 30, 31,  1,  2,  3,  4,  5])

In [25]:
#구매시각에서 1시간 단위로 시간대 파생변수 생성
df['구매시각'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 123191 entries, 0 to 123190
Series name: 구매시각
Non-Null Count   Dtype 
--------------   ----- 
123191 non-null  object
dtypes: object(1)
memory usage: 962.6+ KB


In [26]:
df['구매시각']=pd.to_datetime(df['구매시각'],format='%H:%M').dt.time

In [27]:
#시간만 추출하여 시간대 컬럼 생성
df['구매시간대']=df['구매시각'].apply(lambda x : x.hour)

In [29]:
df['구매시간대'].unique()

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,  9,  8])

In [30]:
df['반품일자'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 123191 entries, 0 to 123190
Series name: 반품일자
Non-Null Count  Dtype 
--------------  ----- 
1789 non-null   object
dtypes: object(1)
memory usage: 962.6+ KB


In [31]:
#반품소요기간 = 반품일자 - 공급일자를 뺀 기간
df['반품일자']= pd.to_datetime(df['반품일자']) # 날짜 데이터 형식 변환

In [32]:
df['반품소요기간']=(df['반품일자']-df['공급일자']).dt.days

In [33]:
df.head(1)
#Nan값인데 날짜 형식으로 바꾸려고해서 NaT로 바뀜

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04:00,NaT,1,6,10,NaN


In [40]:
df['반품소요기간'].unique()

array([nan,  3.,  2.,  1.])

In [41]:
df['반품소요기간'].info()

0   NaN
1   NaN
2   NaN
3   NaN
4   NaN
Name: 반품소요기간, dtype: float64

In [44]:
# NaN 값을 0으로 대체
df['반품소요기간'] = df['반품소요기간'].fillna(0).astype(int)
# 반품소요기간 정수형으로 변환
df['반품소요기간']=df['반품소요기간'].astype(int)

In [45]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 123191 entries, 0 to 123190
Data columns (total 16 columns):
 #   Column  Non-Null Count   Dtype         
---  ------  --------------   -----         
 0   공급일자    123191 non-null  datetime64[ns]
 1   회원번호    123191 non-null  int64         
 2   조합원상태   123191 non-null  object        
 3   물품대분류   123191 non-null  object        
 4   물품중분류   123191 non-null  object        
 5   물품소분류   123191 non-null  object        
 6   물품명     123191 non-null  object        
 7   구매수량    123191 non-null  float64       
 8   구매금액    123191 non-null  int64         
 9   구매매장    123191 non-null  object        
 10  구매시각    123191 non-null  object        
 11  반품일자    1789 non-null    datetime64[ns]
 12  구매월     123191 non-null  int64         
 13  구매일     123191 non-null  int64         
 14  구매시간대   123191 non-null  int64         
 15  반품소요기간  123191 non-null  int64         
dtypes: datetime64[ns](2), float64(1), int64(6), object(7)
memory usage: 15.0+ 

In [46]:
df.head()

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,272369856,정상회원,과실,과일,사과,사과/유(1.5kg),1.0,22207,매장C,10:04:00,NaT,1,6,10,0
1,2023-01-06,1476143616,정상회원,간식,빵,"식빵, 식사대용",아침빵(240g/8개)/매장,1.0,4403,매장C,10:09:00,NaT,1,6,10,0
2,2023-01-06,743192512,정상회원,채소,양념채소,생강,생강(300g),1.0,7083,매장C,10:12:00,NaT,1,6,10,0
3,2023-01-06,176912064,정상회원,축산물,유제품,우유/산양유,유기농저지방우유(종이팩/900ml),1.0,7849,매장C,10:23:00,NaT,1,6,10,0
4,2023-01-06,87592064,정상회원,양념/가루,가루,양념가루,카레(매운맛: 100g),3.0,15507,매장C,10:29:00,NaT,1,6,10,0


In [47]:
df.describe()

,회원번호,구매수량,구매금액,구매월,구매일,구매시간대,반품소요기간
count,1.231910e+05,123191.000000,1.231910e+05,123191.000000,123191.000000,123191.000000,123191.000000
mean,7.184267e+08,1.136326,1.079656e+04,5.606676,15.667533,13.974633,0.019620
std,4.357415e+08,0.690563,1.672424e+04,2.866247,8.695271,2.829876,0.179327
min,1.861382e+07,-30.000000,-6.662160e+05,1.000000,1.000000,8.000000,0.000000
25%,3.424877e+08,1.000000,4.020000e+03,3.000000,8.000000,11.000000,0.000000
50%,7.177839e+08,1.000000,7.275000e+03,6.000000,16.000000,14.000000,0.000000
75%,1.052064e+09,1.000000,1.263500e+04,8.000000,23.000000,16.000000,0.000000
max,1.670606e+09,30.000000,1.837838e+06,11.000000,31.000000,20.000000,3.000000


In [48]:
#c매장 csv 파일 추출
# DataFrame을 'my_dataframe.csv' 파일로 저장
df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/C지점.csv', index=False)


# D매장 데이터 전처리

In [49]:
df= pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/product_data_002.csv',encoding='euc-kr')

In [50]:
df.head(1)

,Unnamed: 0.1,Unnamed: 0,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,0,0,2023-01-06,133834304,정상회원,잡곡,콩/화본/깨,화본,귀리(무/500g),1.0,8232,매장D,10:32,NaN


In [51]:
# Unnamed : 0.1 , Unnamed: 0 불필요한 컬럼으로 삭제
df =df.drop(['Unnamed: 0.1', 'Unnamed: 0'],axis=1)
df.head()

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,2023-01-06,133834304,정상회원,잡곡,콩/화본/깨,화본,귀리(무/500g),1.0,8232,매장D,10:32,NaN
1,2023-01-06,749556736,정상회원,잡곡,콩/화본/깨,토종,선비콩/유(500g),1.0,17038,매장D,10:44,NaN
2,2023-01-06,787320768,정상회원,채소,잎/줄기채소,시금치,시금치(300g),1.0,4403,매장D,11:10,NaN
3,2023-01-06,175843008,정상회원,채소,뿌리채소,당근,당근(500g),2.0,5360,매장D,11:45,NaN
4,2023-01-06,165261952,정상회원,간식,밀가공,만두,군만두(500g),1.0,8998,매장D,12:14,NaN


In [52]:
#인덱스 재설정
df=df.reset_index(drop=True)

In [53]:
#공급일자에서 구매월 / 구매일 파생변수 생성
df['공급일자']= pd.to_datetime(df['공급일자'])

In [54]:
df['구매월']=df['공급일자'].dt.month
df['구매일']=df['공급일자'].dt.day

In [55]:
df['구매시각']=pd.to_datetime(df['구매시각'],format='%H:%M').dt.time
#시간만 추출하여 시간대 컬럼 생성
df['구매시간대']=df['구매시각'].apply(lambda x : x.hour)

In [56]:
#반품소요기간 = 반품일자 - 공급일자를 뺀 기간
df['반품일자']= pd.to_datetime(df['반품일자']) # 날짜 데이터 형식 변환

In [58]:
df['반품소요기간']=(df['반품일자']-df['공급일자']).dt.days

In [59]:
# NaN 값을 0으로 대체
df['반품소요기간'] = df['반품소요기간'].fillna(0).astype(int)
# 반품소요기간 정수형으로 변환
df['반품소요기간']=df['반품소요기간'].astype(int)

In [60]:
df.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,133834304,정상회원,잡곡,콩/화본/깨,화본,귀리(무/500g),1.0,8232,매장D,10:32:00,NaT,1,6,10,0


In [61]:
#D매장 csv 파일 추출
# DataFrame을 'my_dataframe.csv' 파일로 저장
df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/D지점.csv', index=False)


# B매장 데이터 전처리

In [62]:
df = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/product_data_003.csv',encoding='euc-kr')

In [63]:
df.head(1)

,Unnamed: 0.1,Unnamed: 0,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,0,0,2023-01-06,345017408,정상회원,양념/가루,장,장류/기타,청국장(150g),1.0,4786,매장B,10:20,NaN


In [64]:
# Unnamed : 0.1 , Unnamed: 0 불필요한 컬럼으로 삭제
df =df.drop(['Unnamed: 0.1', 'Unnamed: 0'],axis=1)

#인덱스 재설정
df=df.reset_index(drop=True)

#공급일자에서 구매월 / 구매일 파생변수 생성
df['공급일자']= pd.to_datetime(df['공급일자'])

df['구매월']=df['공급일자'].dt.month
df['구매일']=df['공급일자'].dt.day

df['구매시각']=pd.to_datetime(df['구매시각'],format='%H:%M').dt.time
#시간만 추출하여 시간대 컬럼 생성
df['구매시간대']=df['구매시각'].apply(lambda x : x.hour)

#반품소요기간 = 반품일자 - 공급일자를 뺀 기간
df['반품일자']= pd.to_datetime(df['반품일자']) # 날짜 데이터 형식 변환

df['반품소요기간']=(df['반품일자']-df['공급일자']).dt.days

# NaN 값을 0으로 대체
df['반품소요기간'] = df['반품소요기간'].fillna(0).astype(int)
# 반품소요기간 정수형으로 변환
df['반품소요기간']=df['반품소요기간'].astype(int)

In [65]:
df.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,345017408,정상회원,양념/가루,장,장류/기타,청국장(150g),1.0,4786,매장B,10:20:00,NaT,1,6,10,0


In [66]:
#B매장 csv 파일 추출
# DataFrame을 'my_dataframe.csv' 파일로 저장
df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/B지점.csv', index=False)

# A매장 데이터 전처리

In [67]:
df = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/product_data_004.csv',encoding='euc-kr')

In [68]:
df.head(1)

,Unnamed: 0.1,Unnamed: 0,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자
0,0,0,2023-01-06,220954944,정상회원,반찬,두부/유부,두부,두부(1모:420g),1.0,3829,매장A,10:32,NaN


In [69]:
# Unnamed : 0.1 , Unnamed: 0 불필요한 컬럼으로 삭제
df =df.drop(['Unnamed: 0.1', 'Unnamed: 0'],axis=1)

#인덱스 재설정
df=df.reset_index(drop=True)

#공급일자에서 구매월 / 구매일 파생변수 생성
df['공급일자']= pd.to_datetime(df['공급일자'])

df['구매월']=df['공급일자'].dt.month
df['구매일']=df['공급일자'].dt.day

df['구매시각']=pd.to_datetime(df['구매시각'],format='%H:%M').dt.time
#시간만 추출하여 시간대 컬럼 생성
df['구매시간대']=df['구매시각'].apply(lambda x : x.hour)

#반품소요기간 = 반품일자 - 공급일자를 뺀 기간
df['반품일자']= pd.to_datetime(df['반품일자']) # 날짜 데이터 형식 변환

df['반품소요기간']=(df['반품일자']-df['공급일자']).dt.days

# NaN 값을 0으로 대체
df['반품소요기간'] = df['반품소요기간'].fillna(0).astype(int)
# 반품소요기간 정수형으로 변환
df['반품소요기간']=df['반품소요기간'].astype(int)

In [70]:
df.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,220954944,정상회원,반찬,두부/유부,두부,두부(1모:420g),1.0,3829,매장A,10:32:00,NaT,1,6,10,0


In [71]:
#A매장 csv 파일 추출
# DataFrame을 'my_dataframe.csv' 파일로 저장
df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/A지점.csv', index=False)

# 지점 A,B,C,D 데이터 통합

In [73]:
A = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/A지점.csv')

In [74]:
A.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,220954944,정상회원,반찬,두부/유부,두부,두부(1모:420g),1.0,3829,매장A,10:32:00,NaN,1,6,10,0


In [75]:
B = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/B지점.csv')
C = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/C지점.csv')
D = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/D지점.csv')

In [78]:
D.head(1)

,공급일자,회원번호,조합원상태,물품대분류,물품중분류,물품소분류,물품명,구매수량,구매금액,구매매장,구매시각,반품일자,구매월,구매일,구매시간대,반품소요기간
0,2023-01-06,133834304,정상회원,잡곡,콩/화본/깨,화본,귀리(무/500g),1.0,8232,매장D,10:32:00,NaN,1,6,10,0


In [79]:
# A, B, C, D를 순서대로 수직 결합
combined_df = pd.concat([A, B, C, D], ignore_index=True)

In [80]:
combined_df['구매매장'].unique()

array(['매장A', '매장B', '매장C', '매장D'], dtype=object)

In [81]:
combined_df.shape # 총 29만 4645개 데이터 16개 컬럼

(294645, 16)

In [82]:
#전체 매장 데이터 csv 변환
combined_df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/전체매장데이터.csv', index=False)

# 멤버 데이터 확인

In [ ]:
#파일 불러오기
mem_info = pd.read_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/member_data.csv')
mem_info.head()

,회원번호,조합원상태,주소,성별,연령,배송서비스신청여부,모바일알람여부,Gold_member
0,1000012800,정상회원,수지구-풍덕천동,여,58.0,미신청,.,normal
1,1000103744,정상회원,처인구-고림동,여,34.0,미신청,.,normal
2,1000122304,정상회원,광주-기타,여,38.0,미신청,.,normal
3,1000163136,정상회원,수지구-동천동,여,73.0,미신청,.,normal
4,1000281920,정상회원,수지구-풍덕천동,여,42.0,신청,.,normal


In [ ]:
#매장 csv 파일 추출
# DataFrame을 'my_dataframe.csv' 파일로 저장
df.to_csv('/content/drive/MyDrive/최강<포빅아A4>/마켓프로젝트/마켓 CSV파일모음/D지점.csv', index=False)

In [ ]:
df

In [ ]:
#데이터 정보 확인
mem_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11555 entries, 0 to 11554
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   회원번호         11555 non-null  int64  
 1   조합원상태        11555 non-null  object 
 2   주소           11555 non-null  object 
 3   성별           11553 non-null  object 
 4   연령           11553 non-null  float64
 5   배송서비스신청여부    11555 non-null  object 
 6   모바일알람여부      11555 non-null  object 
 7   Gold_member  11555 non-null  object 
dtypes: float64(1), int64(1), object(6)
memory usage: 722.3+ KB


In [ ]:
# 회원 정보 총 11555개 , 8개 컬럼
mem_info.shape

(11555, 8)

In [ ]:
# 컬럼명 영어로 변환
# Assuming your DataFrame is named df
mem_info.rename(columns={
    '회원번호': 'member_id',
    '조합원상태': 'membership_status',
    '주소': 'address',
    '성별': 'gender',
    '연령': 'age',
    '배송서비스신청여부': 'delivery_service_status',
    '모바일알람여부': 'mobile_alert_status',
    'Gold_member': 'gold_member_status'
}, inplace=True)

In [ ]:
mem_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11555 entries, 0 to 11554
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   member_id                11555 non-null  int64  
 1   membership_status        11555 non-null  object 
 2   address                  11555 non-null  object 
 3   gender                   11553 non-null  object 
 4   age                      11553 non-null  float64
 5   delivery_service_status  11555 non-null  object 
 6   mobile_alert_status      11555 non-null  object 
 7   gold_member_status       11555 non-null  object 
dtypes: float64(1), int64(1), object(6)
memory usage: 722.3+ KB


## 각 컬럼 고유값(범주) 확인

In [ ]:
# 컬럼 타입에 따라 정보 출력
for column in mem_info.columns:
    print(f"{column} 컬럼:")
    if column == 'member_id':
        print(mem_info[column].nunique())
    elif mem_info[column].dtype == 'object':
        print(mem_info[column].value_counts())
    else:
        print(mem_info[column].describe())
    print()

member_id 컬럼:
11555

membership_status 컬럼:
정상회원     11413
탈퇴         139
탈퇴처리중        2
탈퇴신청         1
Name: membership_status, dtype: int64

address 컬럼:
수지구-풍덕천동    3188
수지구-상현동     1601
수지구-신봉동     1422
기타-기타       1126
수지구-동천동      924
            ... 
수정구-복정동        1
수정구-상대원동       1
기타-비조합원        1
기타-동천동         1
기타-성남동         1
Name: address, Length: 95, dtype: int64

gender 컬럼:
여    10885
남      668
Name: gender, dtype: int64

age 컬럼:
count    11553.000000
mean        49.704233
std         11.563223
min         17.000000
25%         41.000000
50%         48.000000
75%         56.000000
max        108.000000
Name: age, dtype: float64

delivery_service_status 컬럼:
미신청    10512
신청      1043
Name: delivery_service_status, dtype: int64

mobile_alert_status 컬럼:
.     10653
수신      902
Name: mobile_alert_status, dtype: int64

gold_member_status 컬럼:
normal    11350
VIP         205
Name: gold_member_status, dtype: int64



In [ ]:
# 가입 회원 수
print("회원번호 컬럼:")
print(mem_info['member_id'].nunique())  # 회원번호는 고유값이므로 개별 값의 빈도를 출력


회원번호 컬럼:
11555


In [ ]:
#회원 주소 구/동 분리
# address 컬럼을 기준으로 구(gu)와 동(dong) 컬럼으로 분할
mem_info[['gu', 'dong']] = mem_info['address'].str.split('-', expand=True)

# 분할된 컬럼들을 확인
mem_info.head()

,member_id,membership_status,address,gender,age,delivery_service_status,mobile_alert_status,gold_member_status,gu,dong
0,1000012800,정상회원,수지구-풍덕천동,여,58.0,미신청,.,normal,수지구,풍덕천동
1,1000103744,정상회원,처인구-고림동,여,34.0,미신청,.,normal,처인구,고림동
2,1000122304,정상회원,광주-기타,여,38.0,미신청,.,normal,광주,기타
3,1000163136,정상회원,수지구-동천동,여,73.0,미신청,.,normal,수지구,동천동
4,1000281920,정상회원,수지구-풍덕천동,여,42.0,신청,.,normal,수지구,풍덕천동


In [ ]:
# 구 확인
print('구 종류 : ' , mem_info['gu'].unique())
print('구 개수 : ',mem_info['gu'].nunique())

구 종류 :  ['수지구' '처인구' '광주' '기흥구' '기타' '분당구' '수원 영통구' '서울 강남구' '수원 팔달구' '서울 송파구'
 '화성' '중원구' '수원 장안구' '수정구' '수원 권선구' '하남']
구 개수 :  16


In [ ]:
#구가 '기타'
len(mem_info[mem_info['gu'] == '기타'])

1149

In [ ]:
# 동 확인
print('동 종류 : ' , mem_info['dong'].unique())
print('동 개수 : ',mem_info['dong'].nunique())

동 종류 :  ['풍덕천동' '고림동' '기타' '동천동' '신봉동' '상하동' '상현동' '성복동' '백현동' '판교동' '죽전동' '구미동'
 '동백동' '보정동' '청덕동' '수내동' '언남동' '중동' '삼평동' '마북동' '오포읍' '정자동' '금광동' '역북동'
 '삼가동' '신흥동' '서현동' '야탑동' '보라동' '구갈동' '김량장동' '이동면' '분당동' '신갈동' '영덕동'
 '위례(창곡)동' '모현면' '원삼면' '이매동' '금곡동' '공세동' '상갈동' '복정동' '서천동' '산성동' '도촌동'
 '유방동' '남사면' '은행동' '궁내동' '성남동' '여수동' '비조합원' '상대원동' '하대원동' '중앙동' '단대동' '율동'
 '지곡동' '마평동' '태평동' '위례동']
동 개수 :  62


In [ ]:
#동이 '기타'
len(mem_info[mem_info['dong'] == '기타'])

1925

구 - 기타 1149, 동 - 기타 1925개인 것으로 보아 구는 기입했지만 동은 기입 안된 정보가 있는 것 같음

## 결측치 확인

In [ ]:
#결측치 확인
mem_info.isnull().sum()

member_id                  0
membership_status          0
address                    0
gender                     2
age                        2
delivery_service_status    0
mobile_alert_status        0
gold_member_status         0
gu                         0
dong                       0
dtype: int64

In [ ]:
# 결측치가 있는 행 보기
mem_info[mem_info['gender'].isnull()]

,member_id,membership_status,address,gender,age,delivery_service_status,mobile_alert_status,gold_member_status,gu,dong
3666,1527053696,정상회원,기타-기타,NaN,45.0,미신청,.,normal,기타,기타
7098,545864448,정상회원,기타-기타,NaN,NaN,미신청,.,normal,기타,기타


In [ ]:
# 결측치가 있는 행 보기
mem_info[mem_info['age'].isnull()]

,member_id,membership_status,address,gender,age,delivery_service_status,mobile_alert_status,gold_member_status,gu,dong
5983,383599936,정상회원,기타-기타,여,NaN,미신청,.,normal,기타,기타
7098,545864448,정상회원,기타-기타,NaN,NaN,미신청,.,normal,기타,기타


# 각 컬럼 분포(파이차트) 확인

In [ ]:
import matplotlib as mpl

plt.rcParams['font.family'] = 'NanumGothic'
mpl.rcParams['axes.unicode_minus'] = False

In [ ]:
import matplotlib.pyplot as plt

# 한글 폰트 설정
plt.rcParams['font.family'] = 'NanumGothic'

# 각 컬럼에 대해 개별적으로 파이차트 그리기
categorical_columns = [col for col in mem_info.columns if mem_info[col].dtype == 'object']

for column in categorical_columns:
    plt.figure(figsize=(8, 6))
    plt.title(f"Distribution of {column}")

    # 각 범주의 빈도 계산하여 파이차트 그리기
    mem_info[column].value_counts().plot.pie(autopct='%1.1f%%', startangle=140)
    plt.ylabel('')
    plt.show()


NameError: name 'mem_info' is not defined